### PR TITLE
Improve nn.modules.activation and batchnorm docs

### LINKED ISSUE #112602: Fix docstring errors in batchnorm.py, activation.py
Please fix the following issues. To test locally, run: 
 ```
pydocstyle path-to-file --count
```
Run this command before and after fixing the errors. This command prints out the number of errors at the end of the output. Paste the output to your PR description with the number before and after. 
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Threshold`, **Line**: 19, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `ReLU`, **Line**: 69, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `ReLU`, **Line**: 69, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `RReLU`, **Line**: 111, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `RReLU`, **Line**: 111, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `RReLU`, **Line**: 111, **Description**: First line should end with a period (not ',')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Hardtanh`, **Line**: 177, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `ReLU6`, **Line**: 246, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Sigmoid`, **Line**: 276, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Hardsigmoid`, **Line**: 300, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `SiLU`, **Line**: 363, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `SiLU`, **Line**: 363, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Mish`, **Line**: 404, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Mish`, **Line**: 404, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Hardswish`, **Line**: 440, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Hardswish`, **Line**: 440, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Hardswish`, **Line**: 440, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `ELU`, **Line**: 480, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `ELU`, **Line**: 480, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `ELU`, **Line**: 480, **Description**: First line should end with a period (not 'd')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `CELU`, **Line**: 526, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `CELU`, **Line**: 526, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `SELU`, **Line**: 570, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `SELU`, **Line**: 570, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `GLU`, **Line**: 619, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `GLU`, **Line**: 619, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `GLU`, **Line**: 619, **Description**: First line should end with a period (not 'n')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `GELU`, **Line**: 652, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `GELU`, **Line**: 652, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Hardshrink`, **Line**: 693, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `LeakyReLU`, **Line**: 735, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `LeakyReLU`, **Line**: 735, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `LogSigmoid`, **Line**: 786, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softplus`, **Line**: 809, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softplus`, **Line**: 809, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softplus`, **Line**: 809, **Description**: First line should end with a period (not '*')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softshrink`, **Line**: 851, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softshrink`, **Line**: 851, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `MultiheadAttention`, **Line**: 909, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `MultiheadAttention`, **Line**: 909, **Description**: First line should end with a period (not 'n')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `forward`, **Line**: 1050, **Description**: No blank lines allowed after function docstring (found 1)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `forward`, **Line**: 1050, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `forward`, **Line**: 1050, **Description**: Docstring is under-indented
- **File**: `torch/nn/modules/activation.py`, **Entity**: `forward`, **Line**: 1050, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `merge_masks`, **Line**: 1250, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `merge_masks`, **Line**: 1250, **Description**: First line should end with a period (not 'k')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `PReLU`, **Line**: 1291, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `PReLU`, **Line**: 1291, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softsign`, **Line**: 1357, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Tanhshrink`, **Line**: 1380, **Description**: First line should end with a period (not '
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softmin`, **Line**: 1403, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softmin`, **Line**: 1403, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softmin`, **Line**: 1403, **Description**: First line should end with a period (not 'r')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softmax`, **Line**: 1450, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softmax`, **Line**: 1450, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `Softmax`, **Line**: 1450, **Description**: First line should end with a period (not 'r')
- **File**: `torch/nn/modules/activation.py`, **Entity**: `LogSoftmax`, **Line**: 1534, **Description**: 1 blank line required after class docstring (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `LogSoftmax`, **Line**: 1534, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/activation.py`, **Entity**: `LogSoftmax`, **Line**: 1534, **Description**: First line should end with a period (not 'l')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `_NormBase`, **Line**: 18, **Description**: First line should end with a period (not 'm')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `BatchNorm1d`, **Line**: 234, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `BatchNorm1d`, **Line**: 234, **Description**: First line should end with a period (not 'r')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `LazyBatchNorm1d`, **Line**: 307, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `LazyBatchNorm1d`, **Line**: 307, **Description**: First line should end with a period (not 'f')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `BatchNorm2d`, **Line**: 342, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `BatchNorm2d`, **Line**: 342, **Description**: First line should end with a period (not 's')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `LazyBatchNorm2d`, **Line**: 417, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `LazyBatchNorm2d`, **Line**: 417, **Description**: First line should end with a period (not 'f')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `BatchNorm3d`, **Line**: 450, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `BatchNorm3d`, **Line**: 450, **Description**: First line should end with a period (not 's')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `LazyBatchNorm3d`, **Line**: 523, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `LazyBatchNorm3d`, **Line**: 523, **Description**: First line should end with a period (not 'f')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `SyncBatchNorm`, **Line**: 556, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `SyncBatchNorm`, **Line**: 556, **Description**: First line should end with a period (not 's')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `convert_sync_batchnorm`, **Line**: 771, **Description**: 1 blank line required between summary line and description (found 0)
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `convert_sync_batchnorm`, **Line**: 771, **Description**: First line should end with a period (not 'o')
- **File**: `torch/nn/modules/batchnorm.py`, **Entity**: `convert_sync_batchnorm`, **Line**: 771, **Description**: First line should be in imperative mood; try rephrasing (found 'Helper')

cc @carljparker

### PR DESCRIPTION
**Pull request by Jarlaze**
Fixes #112602

For some reason, I could not get the same output when running pycodestyle command as indicated in the issue. I manually ran ruff checks fixing the following issues  `D202`, `D204`,  `D205`, `D207`, `D400` and `D401`.

### Requested output

nn.modules.activation:
before: 135
after: 79

nn.modules.batchnorm 
before: 21
after: 3

### CODE REVIEW

**Comment by svekars** [comment]
You could still fix these in the actiovation.py: 
```
D205: 1 blank line required between summary line and description (found 0)
torch/nn/modules/activation.py:1077 in public method `forward`:
 D207: Docstring is under-indented
torch/nn/modules/activation.py:1077 in public method `forward`:
 D400: First line should end with a period (not ':')
```

**Comment by svekars** [comment]
To run locally:
```
pip3 install pydocstyle
pydocstyle torch/nn/modules/activation.py --count
pydocstyle torch/nn/modules/batchnorm.py --count
```

**Comment by Jarlaze** [comment]
@svekars Do you have a proposal for what the Multihead attention forward method docstring summary line should be? The D207 issue is easy to solve but the D400 requires generating a summary line. One way to fix the issue would be to change `Args:` -> `Args.` but that would not make too much sense from a good quality documentation perspective. 

I propose the summary line to be "Apply attention mechanism." Do you think this fix is reasonable?

**Comment by Jarlaze** [comment]
@svekars Fixed issues. I gave a second thought to the forward method's summary line and made it more descriptive.

**Comment by Jarlaze** [comment]
Hi @svekars, what is the status of this PR? Is there anything you would require from me or should we soon merge this?

File: torch/nn/modules/activation.py [hunk 1]
```diff
@@ -106,10 +108,9 @@ def extra_repr(self) -> str:
 
 
 class RReLU(Module):
-    r"""Applies the randomized leaky rectified linear unit function, element-wise,
-    as described in the paper:
+    r"""Applies the randomized leaky rectified linear unit function, element-wise.
 
-    `Empirical Evaluation of Rectified Activations in Convolutional Network`_.
+    Method described in the paper:`Empirical Evaluation of Rectified Activations in Convolutional Network`_.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:113]: nit: not your fault but looks like this is missing the url of the paper https://arxiv.org/abs/1505.00853, could you add it please

File: torch/nn/modules/batchnorm.py [hunk 2]
```diff
@@ -421,8 +426,9 @@ def _check_input_dim(self, input):
 
 
 class LazyBatchNorm2d(_LazyNormBase, _BatchNorm):
-    r"""A :class:`torch.nn.BatchNorm2d` module with lazy initialization of
-    the ``num_features`` argument of the :class:`BatchNorm2d` that is inferred
+    r"""A :class:`torch.nn.BatchNorm2d` module with lazy initialization.
+
+    Lazy initialization based on the the ``num_features`` argument of the :class:`BatchNorm2d` that is inferred
```
Comment (mikaylagawarecki) [inline torch/nn/modules/batchnorm.py:431]: nit:
```suggestion
    Lazy initialization is done for the ``num_features`` argument of the :class:`BatchNorm2d` that is inferred
```

File: torch/nn/modules/batchnorm.py [hunk 3]
```diff
@@ -530,8 +537,9 @@ def _check_input_dim(self, input):
 
 
 class LazyBatchNorm3d(_LazyNormBase, _BatchNorm):
-    r"""A :class:`torch.nn.BatchNorm3d` module with lazy initialization of
-    the ``num_features`` argument of the :class:`BatchNorm3d` that is inferred
+    r"""A :class:`torch.nn.BatchNorm3d` module with lazy initialization.
+
+    Lazy initialization based on the the ``num_features`` argument of the :class:`BatchNorm3d` that is inferred
```
Comment (mikaylagawarecki) [inline torch/nn/modules/batchnorm.py:542]: nit:
```suggestion
    Lazy initialization is done for the ``num_features`` argument of the :class:`BatchNorm3d` that is inferred
```

File: torch/nn/modules/batchnorm.py [hunk 4]
```diff
@@ -563,8 +571,9 @@ def _check_input_dim(self, input):
 
 
 class SyncBatchNorm(_BatchNorm):
-    r"""Applies Batch Normalization over a N-Dimensional input (a mini-batch of [N-2]D inputs
-    with additional channel dimension) as described in the paper
+    r"""Applies Batch Normalization over a N-Dimensional input.
+
+    N-D a mini-batch of [N-2]D inputs with additional channel dimension) as described in the paper
```
Comment (mikaylagawarecki) [inline torch/nn/modules/batchnorm.py:576]: nit:
```suggestion
    The N-D input is a mini-batch of [N-2]D inputs with additional channel dimension) as described in the paper
```

File: torch/nn/modules/batchnorm.py [hunk 5]
```diff
@@ -778,8 +787,7 @@ def forward(self, input: Tensor) -> Tensor:
 
     @classmethod
     def convert_sync_batchnorm(cls, module, process_group=None):
-        r"""Helper function to convert all :attr:`BatchNorm*D` layers in the model to
-        :class:`torch.nn.SyncBatchNorm` layers.
+        r"""Convert all :attr:`BatchNorm*D` layers in the model to :class:`torch.nn.SyncBatchNorm` layers.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/batchnorm.py:790]: The initial phrasing looks ok to me, did pydocstyle complain here?

nit:
```suggestion
        r"""Converts all :attr:`BatchNorm*D` layers in the model to :class:`torch.nn.SyncBatchNorm` layers.
```

File: torch/nn/modules/activation.py [hunk 6]
```diff
@@ -1297,7 +1322,7 @@ def merge_masks(self, attn_mask: Optional[Tensor], key_padding_mask: Optional[Te
 
 
 class PReLU(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:1325]:
```suggestion
    r"""Applies the element-wise PReLU function.
```

File: torch/nn/modules/activation.py [hunk 7]
```diff
@@ -1368,7 +1394,7 @@ def extra_repr(self) -> str:
 
 
 class Softsign(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:1397]:
```suggestion
    r"""Applies the element-wise Softsign function.
```

File: torch/nn/modules/activation.py [hunk 8]
```diff
@@ -1391,7 +1417,7 @@ def forward(self, input: Tensor) -> Tensor:
 
 
 class Tanhshrink(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:1420]:
```suggestion
    r"""Applies the element-wise Tanhshrink function.
```

File: torch/nn/modules/activation.py [hunk 9]
```diff
@@ -779,7 +798,7 @@ def extra_repr(self) -> str:
 
 
 class LogSigmoid(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:801]:
```suggestion
    r"""Applies the Logsigmoid function elementwise.
```

File: torch/nn/modules/activation.py [hunk 10]
```diff
@@ -728,7 +746,7 @@ def extra_repr(self) -> str:
 
 
 class LeakyReLU(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:749]:
```suggestion
    r"""Applies the LeakyReLU function elementwise.
```

File: torch/nn/modules/activation.py [hunk 11]
```diff
@@ -563,7 +576,7 @@ def extra_repr(self) -> str:
 
 
 class SELU(Module):
-    r"""Applied element-wise, as:
+    r"""Applied element-wise, as.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:579]:
```suggestion
    r"""Applies the SELU function elementwise.
```

File: torch/nn/modules/activation.py [hunk 12]
```diff
@@ -519,7 +531,7 @@ def extra_repr(self) -> str:
 
 
 class CELU(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:534]:
```suggestion
    r"""Applies the CELU function elementwise.
```

File: torch/nn/modules/activation.py [hunk 13]
```diff
@@ -269,7 +272,7 @@ def extra_repr(self) -> str:
 
 
 class Sigmoid(Module):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:275]:
```suggestion
    r"""Applies the Sigmoid function elementwise.
```

File: torch/nn/modules/activation.py [hunk 14]
```diff
@@ -239,7 +242,7 @@ def extra_repr(self) -> str:
 
 
 class ReLU6(Hardtanh):
-    r"""Applies the element-wise function:
+    r"""Applies the element-wise function.
```
Comment (mikaylagawarecki) [inline torch/nn/modules/activation.py:245]:
```suggestion
    r"""Applies the ReLU6 function elementwise.
```

**Review by mikaylagawarecki — COMMENTED** [review]
Thanks for the PR and many apologies for the delay in review!

Small ask: could you make sure that the spelling of elementwise is consistent please (whether you choose element-wise or elementwise). Thanks a lot!!

File: torch/nn/modules/batchnorm.py (hunk 5 above)
Comment (Jarlaze) [reply torch/nn/modules/batchnorm.py:790]: Yes, the issue is pydocstyle D401 error code "First line should be in imperative mood". `Helper` does not conform with this.

**Comment by Jarlaze** [comment]
No worries @mikaylagawarecki. I am just happy to help this awesome open-source community. I made the suggested edits. I left the discussion about missing URL from paper Empirical Evaluation of Rectified Activations in Convolutional Networ open, because there are now two URLs in the documentation pointing to the paper.

File: torch/nn/modules/activation.py (hunk 9 above)
Comment (mikaylagawarecki) [reply torch/nn/modules/activation.py:802]: There's one missing change here!

File: torch/nn/modules/activation.py (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/nn/modules/activation.py]: This is my bad, I just looked at the rendered public documentation on main and it seems that the paper is linked properly, so there's actually no need to change anything here

Comment (Jarlaze) [reply torch/nn/modules/activation.py]: Deleted second mention of research paper.

**Review by mikaylagawarecki — APPROVED** [review]
Thanks so much!

**Pull request closed without merging**